connect/runtime/src/main/java/org/apache/kafka/connect/runtime/TransformationChain.java

```diff
@@ -37,9 +41,11 @@ public TransformationChain(List<Transformation<R>> transformations, RetryWithTol
     public R apply(R record) {
         if (transformations.isEmpty()) return record;
 
-        for (final Transformation<R> transformation : transformations) {
+        for (Transformation<R> transformation : transformations) {
```

Comment:
why remove the `final` here?

Reply:
@ewencp I suggested not using `final` in every new loop for consistency (several loops even here don't use it such as the one in `close`), but I didn't imply that we should change unaffected lines. 

In general in Connect my understanding is that we are not strict in demanding use of `final` in local variables. Let me know if something changed.

Reply:
Originally I had added `final` to the new code for-loop of the `toString()` method of the same file, and then removed it. I agreed with @kkonstantine's suggestion, and removed `final` from all the small for-loops of this class (to match the one already in `close()`. It looks like I stumbled into an inconsistency, already in the code, so I took out the change here as you suggest in order to keep this diff small and not conflate cleaning up consistency of "final" keyword with this logging change

Reply:
@kkonstantine I don't feel strongly about it, but others do. Probably largely due to (with hindsight) the opposite default of what would have been ideal, i.e. would have been better to have to opt out of by-default immutable references rather than opt in.

We definitely already have some divergence within the broader AK codebase (`streams/` uses `final` more aggressively, not sure if it's checkstyle-enforced, Connect tends to use `final` mostly just for class fields). `clients` is a bit of a mix, but less `final` in `for` loops than not. And `core` is scala, so different story altogether.

I have no hard and fast rule here, main reason I even asked is because in this diff it is being removed from what was already there. No need to get hung up on it, but the other argument is to not change it in favor of minimizing diffs and maximizing mergability/cherry-pickability :) There are tradeoffs no matter which direction you choose....